lib/index.js

```diff
@@ -150,6 +150,10 @@ function getOptions(options) {
   var success = options.success;
 
   options.error = function(err) {
+    var payload = util._extend(new Error(), JSON.parse(err));
+    if (cb) {
+      cb.call(options.context, payload, null);
+    }
     if (error) {
       error.call(options.context, util._extend(new Error(), JSON.parse(err)));
```

Comment:
Note: TravisCI build is failing. With this change, the magic will happen. :wink: